trainee_template/react_series.amlg

```diff
@@ -325,6 +343,164 @@
 			)
 		)
 
+		;to continue an existing series, pull the necessary last few cases from the series to account for largest lag amount
+		(if continue_series
+			(let
+				(assoc
+					series_id_features (get tsModelFeaturesMap "series_id_features")
+					id_values_map (assoc)
+					existing_series_cases (list)
+				)
+
+				(if
+					(and
+						stop_on_terminator
+						series_has_terminators
+						(not (contains_index initial_features_map tsTimeFeature))
+					)
+					(seq
+						(accum (assoc warnings (assoc "Can't continue terminated series.") ))
+						;immediately stop this series since it's terminated
+						(conclude (conclude
+							(assoc
+								"payload"
+									(assoc
+										"series" (null)
+										"action_features" output_features
+									)
+								"warnings" (if (size warnings) (indices warnings))
+							)
+						))
+					)
+				)
+
+				;populate an assoc of id feature -> value, if conditioned on a series, otherwise select a random series
+				(assign (assoc
+					id_values_map
+						;if initial_features_map has series id features, use them
+						(if (size (filter (unzip initial_features_map series_id_features)))
+							;get ID feature values from initial map
+							(keep initial_features_map series_id_features)
+
+							;else randomly pick IDs by selecting a case at random and using its series ids
+							(zip
+								series_id_features
+								(retrieve_from_entity
+									;pick a case at random
+									(first
+										(contained_entities (list
+											(query_exists internalLabelSession)
+											(query_sample 1)
+										))
+									)
+									;pull the random case's ID features
+									series_id_features
+								)
+							)
+						)
+				))
+
+				;pull the necessary number of cases from the series
+				(assign (assoc
+					existing_series_cases
+						(contained_entities (append
+							(map
+								(lambda (query_equals (target_value) (get id_values_map (target_value))))
+								series_id_features
+							)
+							;if an initial time value is provided, limit existing series cases to those before the specified initial time
+							(if (contains_index initial_features_map tsTimeFeature)
+								(query_less_or_equal_to
+									tsTimeFeature
+									;subtract an epsilon of 1e-6 from the time value so that the query logic
```

Comment:
because when dealing with epoch second values, ie today is: 1,695,268,800  , if subtracting anything smaller than 1e-6, such as 1e-7, the precision is lost and the value won't change. e.g, (- 1695268800 1e-7) will result in 1695268800, but (- 1695268800 1e-6) results in 1695268799.999999. If you have suggestions on how to emulate query_less_than in a more stable way, let me know. Alternatively I can put this story on hold until we actually implement query_less_than in Amalgam and just use that query opcode without this 'hack'.